pkg/observer/data_stats.go

```diff
@@ -0,0 +1,36 @@
+package tracing
+
+import (
+	"github.com/cilium/tetragon/pkg/metrics/consts"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	// Define a counter metric for data event statistics
+	DataEventStats = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name:        consts.MetricNamePrefix + "data_event_stats",
+		Help:        "Data event statistics. For internal use only.",
+		ConstLabels: nil,
+	}, []string{"event", "location"})
+)
+
+type EventType int
+
+const (
+	EventReceived EventType = iota
+	EventMatched
+	EventNotMatched
+)
+
+var EventTypeStrings = map[EventType]string{
+	EventReceived:   "EventReceived",
+	EventMatched:    "EventMatched",
+	EventNotMatched: "EventNotMatched",
```

Comment:
the final output already says it's for data event, so let's keep this just with the action name, like:

jolsa@ubuntu-22:~/tetragon$ curl http://localhost:8080/metrics 2>&1 | grep ^tetragon_data_event_stats
tetragon_data_event_stats{event="Added"} 14
tetragon_data_event_stats{event="Matched"} 14
tetragon_data_event_stats{event="Received"} 14

s/EventReceived/Received/ ...

Reply:
@olsajiri hi sorry for the delay but I am unable to understand what you want to say